# Question 1

In [1]:
import pandas as pd 
import numpy as np

In [2]:
# reads the csv file 'monthly2022.csv' into dataframe monthly without headers and first column as the index
# gets the number of rows in dataframe monthly
monthly = pd.read_csv('monthly2022.csv', index_col=0, header=None)
T = monthly.shape[0]

In [3]:
# getting the columns
basicstats = monthly.describe().transpose() # .describe() gets the count, mean, std, 25th, 50th, and 75th percentiles and transposes them
basicstats = basicstats.drop(['min', 'max'], axis=1) # removes the min and max columns from the basicstats dataframe
basicstats

,count,mean,std,25%,50%,75%
1,1164.0,0.009190,0.053429,-0.018313,0.012866,0.039323
2,1164.0,0.012013,0.071652,-0.020466,0.014148,0.043246
3,1164.0,0.016794,0.103055,-0.029751,0.009320,0.053341
4,1164.0,0.012894,0.088126,-0.026246,0.011118,0.046156
5,1164.0,0.011349,0.079194,-0.022683,0.010806,0.045625
6,1164.0,0.010877,0.074596,-0.024597,0.012692,0.045745
7,1164.0,0.011015,0.071883,-0.022659,0.012491,0.045213
8,1164.0,0.010814,0.069740,-0.022555,0.014197,0.046008
9,1164.0,0.010725,0.066794,-0.022541,0.014180,0.045794
10,1164.0,0.009915,0.062993,-0.021986,0.013715,0.043970


In [4]:
# multiplies 100 to all columns except the count column
basicstats=basicstats.apply(lambda x: x*100 if x.name!='count' else x)
basicstats

,count,mean,std,25%,50%,75%
1,1164.0,0.919010,5.342949,-1.831312,1.286599,3.932300
2,1164.0,1.201292,7.165250,-2.046618,1.414757,4.324634
3,1164.0,1.679443,10.305532,-2.975058,0.932015,5.334085
4,1164.0,1.289393,8.812580,-2.624551,1.111770,4.615560
5,1164.0,1.134927,7.919363,-2.268328,1.080606,4.562462
6,1164.0,1.087679,7.459553,-2.459661,1.269200,4.574530
7,1164.0,1.101484,7.188280,-2.265917,1.249061,4.521342
8,1164.0,1.081399,6.974002,-2.255451,1.419656,4.600824
9,1164.0,1.072540,6.679415,-2.254117,1.418002,4.579421
10,1164.0,0.991528,6.299298,-2.198631,1.371539,4.396950


In [5]:
skew = monthly.skew().rename('skew')
zstat1 = (skew/np.sqrt(6/T)).rename('zstat1') # calculates the z stat for each skew value
kurtosis = monthly.kurtosis().rename('kurtosis')
zstat2 = (kurtosis/np.sqrt(24/T)).rename('zstat2') # calculates the z stat for each kurtosis value

In [6]:
# creates dataframe that compiles all the stat measures into single table
summ_stat=pd.concat([basicstats, skew, zstat1, kurtosis, zstat2], axis=1)

In [7]:
# rounds every value in every column to 3 decimal places excelpt the count column
make_float = lambda x: x if x.name=='count' else round(x, 3)
summ_stat.apply(make_float)

,count,mean,std,25%,50%,75%,skew,zstat1,kurtosis,zstat2
1,1164.0,0.919,5.343,-1.831,1.287,3.932,0.129,1.797,7.585,52.823
2,1164.0,1.201,7.165,-2.047,1.415,4.325,1.443,20.105,14.940,104.045
3,1164.0,1.679,10.306,-2.975,0.932,5.334,2.935,40.879,24.498,170.609
4,1164.0,1.289,8.813,-2.625,1.112,4.616,1.996,27.801,18.635,129.780
5,1164.0,1.135,7.919,-2.268,1.081,4.562,1.912,26.635,21.254,148.015
6,1164.0,1.088,7.460,-2.460,1.269,4.575,1.103,15.358,11.980,83.432
7,1164.0,1.101,7.188,-2.266,1.249,4.521,0.864,12.036,11.004,76.637
8,1164.0,1.081,6.974,-2.255,1.420,4.601,0.615,8.563,8.967,62.449
9,1164.0,1.073,6.679,-2.254,1.418,4.579,0.423,5.887,7.897,54.993
10,1164.0,0.992,6.299,-2.199,1.372,4.397,0.262,3.651,7.955,55.398


In [8]:
# creates jan_only variable that only has rows that are in January and assigns number of rows in jan_only to T
jan_only = monthly.iloc[::12]
T = jan_only.shape[0]

In [9]:
# getting the columns
basicstats = jan_only.describe().transpose()
basicstats = basicstats.drop('max', axis=1)
basicstats=basicstats.apply(lambda x: x*100 if x.name!='count' else x) # multiplies 100 to every value in every column except count column

In [10]:
skew = jan_only.skew().rename('skew')
zstatl = (skew/np.sqrt(6/T)).rename('zstat1')
kurtosis = jan_only.kurtosis().rename('kurtosis')
zstat2 = (kurtosis/np.sqrt(24/T)).rename('zstat2')

In [11]:
# combining all calculations into one dataframe
make_float = lambda x: x if x.name=='count' else round(x, 3)
pd.concat([basicstats, skew, zstat1, kurtosis, zstat2], axis=1).apply(make_float)

,count,mean,std,min,25%,50%,75%,skew,zstat1,kurtosis,zstat2
1,97.0,1.432,4.749,-7.763,-2.621,1.262,4.960,0.279,1.797,-0.072,-0.145
2,97.0,5.152,7.130,-8.290,0.542,4.029,8.897,1.055,20.105,2.224,4.471
3,97.0,12.365,12.598,-8.754,4.144,8.752,15.790,1.586,40.879,3.388,6.811
4,97.0,8.294,9.351,-8.969,2.237,6.305,11.667,1.255,27.801,2.422,4.869
5,97.0,6.675,8.320,-8.920,1.532,5.649,10.050,1.415,26.635,3.778,7.595
6,97.0,5.500,7.899,-10.314,0.647,4.181,8.767,1.298,15.358,3.204,6.441
7,97.0,4.541,7.109,-9.125,-0.163,3.929,7.986,1.018,12.036,2.174,4.371
8,97.0,3.949,7.123,-9.825,-0.770,3.770,8.506,0.934,8.563,2.115,4.251
9,97.0,3.103,6.613,-9.527,-1.061,2.205,7.472,1.006,5.887,2.464,4.954
10,97.0,2.393,5.939,-9.023,-1.863,1.764,6.441,0.651,3.651,1.352,2.718


### Question 1

(a) Compute the average, standard deviation, 25th, 50th, and 75th percentiles based on the monthly returns of the 12 portfolios.

| Portfolio | Mean | Standard Deviation | 25th Percentile | 50th Percentile | 75th Percentile |
| :-: | :-: | :-: | :-: | :-: | :-: |
| 1 | 0.919 | 5.343 | -1.831 | 1.287 | 3.932 |
| 2 | 1.201 | 7.165 | -2.047 | 1.415 | 4.325 |
| 3	| 1.679 | 10.306 | -2.975 | 0.932 |	5.334 |
| 4	| 1.289 | 8.813 | -2.625 | 1.112 | 4.616 |
| 5	| 1.135 | 7.919 | -2.268 | 1.081 | 4.562 |
| 6	| 1.088 | 7.460 | -2.460 | 1.269 | 4.575 |
| 7	| 1.101 | 7.188 | -2.266 | 1.249 | 4.521 |
| 8	| 1.081 | 6.974 | -2.255 | 1.420 | 4.601 |
| 9	| 1.073 | 6.679 | -2.254 | 1.418 | 4.579 |
| 10 | 0.992 | 6.299 | -2.199 | 1.372 |	4.397 |
| 11 | 1.030 | 5.971 | -1.927 | 1.406 |	4.322 |
| 12 | 0.885 | 5.121 | -1.729 | 1.240 | 3.848 |

(b) Compute the coefficients of skewness and (excess) kurtosis based on the monthly returns of the 12 portfolios. For each coefficient, perform a test of normality. What’s your conclusion of the normality of stock returns in the U.S.?

| Portfolio | Skew | Skew Z-stat | Kurtosis | Kurtosis Z-stat |
| :-: | :-: | :-: | :-: | :-: |
| 1 | 0.129 | 1.797 | 7.585 | 52.823 |
| 2	| 1.443 | 20.105 | 14.940 | 104.045 |
| 3	| 2.935 | 40.879 | 24.498 | 170.609 |
| 4	| 1.996 | 27.801 | 18.635 | 129.780 |
| 5	| 1.912 | 26.635 | 21.254 | 148.015 |
| 6 | 1.103 | 15.358 | 11.980 | 83.432 |
| 7	| 0.864 | 12.036 | 11.004 | 76.637 |
| 8	| 0.615 | 8.563 | 8.967 | 62.449 |
| 9	| 0.423 | 5.887 | 7.897 | 54.993 |
| 10 | 0.262 | 3.651 | 7.955 | 55.398 |
| 11 | 0.179 | 2.493 | 7.552 | 52.595 |
| 12 | 0.069 | 0.967 | 7.015 | 48.857 |

Through the above data (skewness, skewness z-stat, kurtosis, kurtosis z-stat), it can be concluded that stock returns in the U.S. are very much not normally distributed. All portfolios show positive skewness and high kurtosis values which means that returns tend to have longer right hand side tails and high likelihood of extreme values.

(c) Repeat the exercises in parts (a) and (b) but based on the returns for the month of January as well as for the rest of year. What’s your conclusion of stock returns in January compared with other months?

| Portfolio | Mean | Standard Deviation | Min | 25th Percentile | 50th Percentile | 75th Percentile | Skew | Skew Z-stat | Kurtosis | Kurtosis Z-stat |
| :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: |
| 1 | 1.432 | 4.749 | -7.763 | -2.621 | 1.262 | 4.960 | 0.279 | 1.797 | -0.072 | -0.145 |
| 2 | 5.152 | 7.130 | -8.290 | 0.542 | 4.029 | 8.897 | 1.055 | 20.105 | 2.224 | 4.471 |
| 3 | 12.365 | 12.598 | -8.754 | 4.144 | 8.752 | 15.790 | 1.586 | 40.879 | 3.388 | 6.811 |
| 4 | 8.294 | 9.351 | -8.969 | 2.237 | 6.305 | 11.667 | 1.255 | 27.801 | 2.422 | 4.869 |
| 5 | 6.675 | 8.320 | -8.920 | 1.532 | 5.649 | 10.050 | 1.415 | 26.635 | 3.778 | 7.595 |
| 6 | 5.500 | 7.899 | -10.314 | 0.647 | 4.181 | 8.767 | 1.298 | 15.358 | 3.204 | 6.441 |
| 7 | 4.541 | 7.109 | -9.125 | -0.163 | 3.929 | 7.986 | 1.018 | 12.036 | 2.174 | 4.371 |
| 8 | 3.949 | 7.123 | -9.825 | -0.770 | 3.770 | 8.506 | 0.934 | 8.563 |	2.115 | 4.251 |
| 9 | 3.103 | 6.613 | -9.527 | -1.061 | 2.205 | 7.472 | 1.006 | 5.887 | 2.464 | 4.954 |
| 10 | 2.393 | 5.939 | -9.023 | -1.863 | 1.764 | 6.441 | 0.651 | 3.651 | 1.352 | 2.718 |
| 11 | 2.096 | 5.462 | -8.549 | -1.772 | 2.148 | 5.604 | 0.650 | 2.493 | 1.365 | 2.744 |
| 12 | 0.989 | 4.576 | -7.959 | -2.422 | 0.821 | 4.379 | 0.192 | 0.967 | -0.265 | -0.533 |

January stock returns seem to show higher average returns as well as median and 75th percentile returns. Standard deviation also seems to be a bit lower overall with kurtosis being lower as well overall. This shows that January was a profitable month compared to the other months and was above average. January was also more stable in performance.

# Question 2

In [12]:
# delete all variables and functions in namespace
%reset -f
# Import the modules
import pandas as pd
import numpy as np 
from datetime import datetime 
from tqdm import tqdm # imports tqdm from the tqdm.notebook library which is progress bar for loops library
import statsmodels.api as sm

In [13]:
# Define the function to calculate the fitted value from recursive OLS
def pred_from_OLS(X, Y, M) :
    """
    Compute the next period predicted value based on recursive OLS regression.
    Args:
        X : The independant variables
        Y : The dependant variable
        M : Initial length of estimation window for out-of-sample tests
    Returns:
        Y_hat: The predicted value based on expanding windows
    """
    
    Y_Hat = np.full((len(Y),1), np.nan)
    Y = np.asarray(Y)
    X = np.asarray(X)
    X = sm.add_constant(X) # adds a constant to the regression model to allow the model to better fit the data

# We start the index at M+1 because the first element of predicted return is at t=M+2.
    for i in range (M+1, len(Y)) :
        Y1 = Y[1:i]
        X1 = X[0:i-1,:]
        reg = sm.OLS(Y1, X1, missing='drop').fit ()
        Y_Hat[i] = reg.predict(X[i-1,:]) # makes a prediction for the i-th observation based on the i-1-th row of X and stores it in Y_Hat
    return Y_Hat

In [14]:
# Define the function to compute two out-of-sample measures of performance
def oos_eval(Y, Pred_Mean, Pred_Var, OOS_SSE_Hist):
    """
    This function computes OOS R^2 and CEV difference based on a vector of predicted mean and variance
    
    Args:
        Y: The excess returns
        Pred_Mean: The time series of predicted mean
        Pred_Var: The time series of predicted variance
        OOS_SSE_Hist: Out-of-sample sum of squares of forecast error based on historical mean
    
    Returns:
        OOS_R2: Out-of-sample R^2
        CE: certainty equivalence of the out-of-sample return
    """
    
    OOS_SSE = np.sum((Y[M+1:]-Pred_Mean[M+1:])**2) # calculates out-of-sample sum of squared errors
    OOS_R2 = 1-OOS_SSE/OOS_SSE_Hist # calculates the out-of-sample R^2
    w0 = ((1/gam)*(Pred_Mean/Pred_Var)).clip(None, 1.5) # calculates the weight on a portfolio
    r0 = Y*w0
    CE = np.mean(r0[M+1:])-gam/2*np.var(r0[M+1:], ddof=1) # calculates the certainty equivalence of the out-of-sample return
    return OOS_R2, CE

In [15]:
# Create the table for in-sample R^2, out-of-sample R^2 and certainty equivlance
def GW2008(X, Y, Y_Hat, M):
    """
    Compute the in-sample and out-of-sample predictability statistics required to reproduce the plots from Goyal & Welch (2008).
    Args:
        X : The independent variables
        Y : The dependant variable
        Y_Hat: The vector of predicted values based on expanding windows
        M : Initial length of estimation window for out-of-sample tests
    Returns:
        data: a dataframe of the statistics of in-sample and out-of-sample tests
    """
    # Create the benchmark using historical average
    Hist_Mean = np.asarray(Y.expanding().mean().shift()) # calculate the expanding mean of Y and shift to avoid look-ahead bias
    Hist_Variance = np.asarray(Y.expanding().var().shift()) # calculate the expanding variance of Y and shift to avoid look-ahead bias
    Y = np.asarray(Y)
    X = np.asarray(X)
    Y_Hat = np.asarray(Y_Hat)
    X = sm.add_constant(X)

    # I. In sample
    Y_1 = Y[1:]
    X_1 = X[0:len(Y)-1,:]
    reg = sm.OLS(Y_1, X_1, missing='drop').fit()
    
    # I.1 OLS Model (non truncated)
    y1 = Y_1-np.mean(Y_1)
    e1 = Y_1-reg.predict(X_1)
    IS_R2 = 1-np.sum(e1**2)/np.sum(y1**2)

    # I.2 OLS Model (truncated)
    e1 = Y_1-reg.predict(X_1).clip(0, None)
    IS_R2_T = 1-np.sum(e1**2)/np.sum(y1**2)

    # II. Out of sample
    # II.1 Historical mean model
    OOS_SSE_Hist = np.sum((Y[M+1:]-Hist_Mean[M+1:])**2)
    # Benchmark Certainty Equivalence
    w0 = ((1/gam) * (Hist_Mean/Hist_Variance)).clip(None, 1.5);
    r0 = Y*w0
    CE0 = np.mean(r0[M+1:])-gam/2*np.var(r0[M+1:], ddof=1)

    # II.2 Forecast based on regression with expanding window
    OOS_R2, CE1 = oos_eval(Y, Y_Hat, Hist_Variance, OOS_SSE_Hist) # calculate out-of-sample R^2 and certainty equivalence 

    # II.3 Trunacated forecast based on regression with expanding window
    OOS_R2_T, CE2 = oos_eval(Y, Y_Hat.clip(0,None), Hist_Variance, OOS_SSE_Hist)

    # creates dataframe to summarize key performance metrics
    data = pd.DataFrame({r'$R_{IS}^2$': IS_R2, 
                     r'$R^2_{IS}$ (Truncated)': IS_R2_T,
                     r'$R^2_{OS}$': OOS_R2,
                     r'$R^2_{OS}$ (Truncated)': OOS_R2_T, 
                     r'$\Delta CEV$': CE1 - CE0,
                     r'$\Delta CEV$ (Truncated)': CE2 - CE0}, 
                    index=[indep_var]) * 100
    return data

In [16]:
def Pool_Avg(Y, Y_Hat, M):
    """
    Compute the out-of-sample predictability statistics based on pooling average.
    Args :
        Y : The dependant variable
        Y_Hat: The vector of predicted values based on expanding windows
        M: Initial length of estimation window for out-of-sample tests
    Returns:
        data: a data frame of the statistics of the out-of-sample tests
    """
    # Create the benchmark using historical average
    Hist_Mean = np.asarray(Y.expanding().mean().shift())
    Hist_Variance = np.asarray(Y.expanding().var().shift())
    Y = np.asarray(Y)
    Y_Hat = np.asarray(Y_Hat)
    
    # Benchmark (Historical Average)
    OOS_SSE_Hist = np.sum((Y[M+1:]-Hist_Mean[M+1:])**2)
    # Benchmark Certainty Equivalence
    w0 = ((1/gam) * (Hist_Mean/Hist_Variance)).clip(None, 1.5);
    r0 = Y*w0
    CE0 = np.mean(r0[M+1:])-gam/2*np.var(r0[M+1:],ddof=1)

    # Preapre for the pooling average
    Y_Hat_Mean = np.mean(Y_Hat, axis=1)
    Y_Hat_Median = np.full((len(Y)), np.nan)

    for i in range(M+1, len(Y)):
        Y_Hat_Median[i] = np.median(Y_Hat[i, :])
    # Alternative: Y_Hat_Median = np.median(Y_Hat,axis=1), this gives a warning message
    OOS_Error_OLS_Pool_Mean = Y-Y_Hat_Mean
    OOS_Error_OLS_Pool_Median = Y-Y_Hat_Median

    # Combination forecast based on mean of various forecasts
    OOS_R2_Pool_Mean, CE_Pool_Mean = oos_eval(Y, Y_Hat_Mean,
                                               Hist_Variance, OOS_SSE_Hist)
    
    # Combination forecast based on median of various forecasts
    OOS_R2_Pool_Median, CE_Pool_Median = oos_eval(Y, Y_Hat_Median,
                                                   Hist_Variance, OOS_SSE_Hist)
    
    # creates dataframe to summarize key performance metrics
    data = pd.DataFrame({r'$R_{OS}^2$ (Pooled Mean)': OOS_R2_Pool_Mean,
                         r'$\Delta CEV$ (Pooled Mean)': CE_Pool_Mean - CE0,
                         r'$R_{OS}^2$ (Pooled Median)': OOS_R2_Pool_Median,
                         r'$\Delta CEV$ (Pooled Median)': CE_Pool_Median - CE0}, 
                        index=['Combination']) * 100
    return data

In [17]:
# Import Data
df_monthly = pd.read_excel('PredictorData2022.xlsx', sheet_name="Monthly")
# Parse the dates properly
time = [str(d) for d in df_monthly.yyyymm]
df_monthly.index = pd.to_datetime(time, format="%Y%m") # converts the time variable into proper datetime index for the dataframe df_monthly

/Users/anaconda3/lib/python3.12/site-packages/openpyxl/worksheet/header_footer.py:48: UserWarning: Cannot parse header or footer so it will be ignored
  warn("""Cannot parse header or footer so it will be ignored""")


In [18]:
# Variable Construction
df_monthly['ExRet'] = df_monthly['CRSP_SPvw']-df_monthly['Rfree']
df_monthly['DP'] = np.log(df_monthly['D12'])-np.log(df_monthly['Index'])
df_monthly['DY'] = np.log(df_monthly['D12'])-np.log(df_monthly['Index'].shift())
df_monthly['EP'] = np.log(df_monthly['E12'])-np.log(df_monthly['Index'])
df_monthly['DE'] = np.log(df_monthly['D12'])-np.log(df_monthly['E12'])
df_monthly['tms'] = df_monthly['lty']-df_monthly['tbl']
df_monthly['dfr'] = df_monthly['corpr']-df_monthly['ltr']
df_monthly['dfy'] = df_monthly['AAA']-df_monthly['BAA']

# shift inflation data to use the previous month's inflation
df_monthly['infl'] = df_monthly['infl'].shift().copy()

# Construction of dependent and independent variables
dep_var = 'ExRet'
indep_vars = ['DE', 'svar', 'dfr', 'lty', 'ltr', 'infl', 'tms',
              'tbl','dfy', 'DP', 'DY', 'EP', 'b/m', 'ntis']

# filter the DataFrame to include only data from December 1926 onwards
subperiod = df_monthly.index>='1926-12-01'
df = df_monthly[subperiod]
M = 240 # number of monthly observations (20 years)
gam = 3 # scaling factor or hyperparameter for model tuning

In [19]:
# make predictions using OLS for all independent variables except 'DE' and 'tms'
df_pred = pd.DataFrame(pred_from_OLS(df[[x for x in indep_vars
                                         if x not in('DE', 'tms')]],df[dep_var], M))
df_pred.columns = ['All']

# make predictions for each independent variable individually and store them in df_pred
for indep_var in tqdm(indep_vars):
    df_pred[indep_var] = pred_from_OLS(df[indep_var], df[dep_var], M)

100%|███████████████████████████████████████████| 14/14 [00:01<00:00, 11.99it/s]


In [20]:
# Create the Table for the in- and out-of-sample performance
pd.options.display.float_format = '{:,.3f}'.format

# Get rid of d/e and tms to avoid perfect multicollinearity
data = GW2008(df[[x for x in indep_vars if x not in('DE', 'tms')]],
              df[dep_var], df_pred['All'], M)
data.rename(index={data.index[0]:'All'},inplace=True)

# calculate performance metrics for each independent variable and concatenate results to the data DataFrame
for indep_var in indep_vars:
    x = GW2008(df[indep_var], df[dep_var], df_pred[indep_var], M)
    data = pd.concat([data, x])
data

,$R_{IS}^2$,$R^2_{IS}$ (Truncated),$R^2_{OS}$,$R^2_{OS}$ (Truncated),$\Delta CEV$,$\Delta CEV$ (Truncated)
All,2.778,2.713,-12.331,-6.524,-0.564,-0.027
DE,0.008,0.008,-0.655,-0.631,0.006,0.008
svar,0.020,0.020,-0.614,-0.604,-0.054,-0.053
dfr,0.061,0.045,-0.356,-0.278,-0.016,-0.004
lty,0.194,0.188,-0.283,0.470,-0.074,-0.015
ltr,0.224,0.210,-0.334,-0.285,0.004,0.008
infl,0.143,0.024,-0.017,0.007,-0.009,-0.008
tms,0.162,0.156,0.123,0.144,0.051,0.051
tbl,0.329,0.287,0.269,0.459,-0.009,0.001
dfy,0.256,0.256,-0.284,-0.272,-0.055,-0.055


In [21]:
# Create table for combination forecasts
Pool_Avg(df[dep_var], df_pred[indep_vars], M)

,$R_{OS}^2$ (Pooled Mean),$\Delta CEV$ (Pooled Mean),$R_{OS}^2$ (Pooled Median),$\Delta CEV$ (Pooled Median)
Combination,0.639,0.005,0.608,0.021


### Question 2

2.1

| Predictive Variables | $R_{IS}^2$ | $R^2_{OS}$ | $\Delta CEV$ | 
| :-: | :-: | :-: | :-: |
| All | 2.778 | -12.331 | -0.564 | 
| DE | 0.008 | -0.655 | 0.006 | 
| svar | 0.020 | -0.614 | -0.054 | 
| dfr | 0.061 | -0.356 | -0.016 | 
| lty | 0.194 | -0.283 | -0.074 | 
| ltr | 0.224 | -0.334 | 0.004 | 
| infl | 0.143 | -0.017 | -0.009 | 						
| tms | 0.162 | 0.123 | 0.051 | 			
| tbl | 0.329 | 0.269 | -0.009 | 				
| dfy | 0.256 | -0.284 | -0.055 | 					
| DP | 0.310 | -0.272 | -0.116 | 				
| DY | 0.401 | -0.898 | -0.148 | 				
| EP | 0.317 | -1.520 |	-0.098 |
| b/m | 0.645 | -2.138 | -0.203 |
| ntis | 0.436 | -0.785 | 0.036 |	

2.2

| Predictive Variables | $R^2_{IS}$ (Truncated) | $R^2_{OS}$ (Truncated) | $\Delta CEV$ (Truncated) | 
| :-: | :-: | :-: | :-: |
| All | 2.713 | -6.524 | -0.027 | 
| DE | 0.008 | -0.631 | 0.008 | 
| svar | 0.020 | -0.604 | -0.053 | 
| dfr | 0.045 | -0.278 | -0.004 | 
| lty | 0.188 | 0.470 | -0.015 | 
| ltr | 0.210 | -0.285 | 0.008 | 
| infl | 0.024 | 0.007 | -0.008 | 						
| tms | 0.156 | 0.144 | 0.051 | 			
| tbl | 0.287 | 0.459 | 0.001 | 				
| dfy | 0.256 | -0.272 | -0.055 | 					
| DP | 0.308 | 0.064 | -0.087 | 				
| DY | 0.390 | -0.173 | -0.093 | 				
| EP | 0.440 | -0.834 | -0.025 |
| b/m | 0.644 | -1.633 | -0.162 |
| ntis | 0.317 | -0.785 | 0.036 |	

2.3 

|  | $R_{OS}^2$ (Pooled Mean) | $\Delta CEV$ (Pooled Mean) | $R_{OS}^2$ (Pooled Median) | $\Delta CEV$ (Pooled Median) |
| :-: | :-: | :-: | :-: | :-: |
| Combination | 0.639 | 0.005 | 0.608 | 0.021 |